server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1088,8 +1188,75 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    public CheckedConsumer<IndexTimeScriptContext, IOException> getPostParsePhase() {
+        if (this.script == null) {
+            return null;
+        }
+        return c -> this.script.compiledScript.executeAndIndex(
+            c.searchLookup,
+            c.leafReaderContext,
+            0,
+            v -> this.indexValue(c.pc, v)
+        );
+    }
+
     @Override
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), type, ignoreMalformedByDefault, coerceByDefault).init(this);
     }
+
+    private interface NumberMapperScript {
+        void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter);
```

Comment:
I find the name a bit distracting because effectively indexing is not what we do in all cases. do we rather collect values and provide them one by one to the emitter consumer?

Reply:
Yes, `executeAndEmit` is better